pkg/app/AppCrudOperationService.go

```diff
@@ -296,7 +298,31 @@ func (impl AppCrudOperationServiceImpl) GetLabelsByAppIdForDeployment(appId int)
 	}
 	labelsDto := make(map[string]string)
 	for _, label := range labels {
-		labelsDto[label.Key] = label.Value
+		labelKey := label.Key
+		labelValue := label.Value
+
+		// if labelKey or labelValue is empty then don't add in labels
+		if len(labelKey) == 0 || len(labelValue) == 0 {
+			impl.logger.Warnw("Ignoring label to propagate to app level", "labelKey", labelKey, "labelValue", labelValue, "appId", appId)
+			continue
+		}
+
+		// if labelKey is not satisfying the label key criteria don't add in labels
+		// label key must be a 'qualified name' (https://github.com/kubernetes/website/issues/17969)
+		errs := validation.IsQualifiedName(labelKey)
+		if len(errs) > 0 {
+			impl.logger.Warnw("Ignoring label to propagate to app level", "err", errors.New(fmt.Sprintf("Validation error - label key - %s is not satisfying the label key criteria", labelKey)), "appId", appId)
```

Comment:
can we simply write string message instead of creating error object

Reply:
yes, we can

Reply:
done